### PR TITLE
Add u32 type and use it for get_local

### LINKED ISSUE #126: the wast parser should fail if the named offset argument is negative
The wat2wasm tool fails when faced with the following command:

`i32.store offset=-1`

With the following error:

`error: unexpected token offset=-1, expected )`

Currently our implementation allows negative values for named arguments.

### PR DESCRIPTION
**Pull request by johnsudaar**
It's a first step to fix #126 but we should add all the different method signatures.
#104 could be the way to go however some method signatures seems to be wrong too (see https://github.com/WebAssembly/wabt/blob/master/src/opcode.def#L62 it's declared as i32,i32 instead of u32,u32). It might be cleaner to use our own data structure?

### CODE REVIEW

File: src/compiler/AST/index.js [hunk 1]
```diff
@@ -5,21 +5,31 @@ const {
...
-    "reinterpret/f32": ["f32"]
-  };
-  return signatures[name] || [object, object];
+  let opcodeName = name;
+  if (object !== undefined) {
+    opcodeName = object + "." + name;
+  }
+  const sign = Signatures[opcodeName];
+  if (sign == undefined) {
+    // TODO: Uncomment this when br_table and others has been done
```
Comment (ColinEberhardt) [inline src/compiler/AST/index.js]: probably worth referencing https://github.com/xtuc/js-webassembly-interpreter/issues/66 (which I now have!)

**Comment by ColinEberhardt** [comment]
LGTM 👍

File: src/compiler/AST/signatures.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  // br_table: [], (vector representation ?)
+  return: [],
+  call: ["u32"],
+  call_indirect: ["u32"],
+
+  // Parametric Instructions
+  drop: [],
+  select: [],
+
+  // Variaable Instructions
```
Comment (xtuc) [inline src/compiler/AST/signatures.js]: typo here

File: src/compiler/AST/signatures.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+  current_memory: [],
+  grow_memory: [],
+
+  // Numeric Instructions
+  "i32.const": ["i32"],
+  "i64.const": ["i64"],
+  "f32.const": ["f32"],
+  "f64.const": ["f64"],
+
+  // Oh boy
```
Comment (xtuc) [inline src/compiler/AST/signatures.js]: lol what is this?

File: src/compiler/AST/signatures.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  "f64.convert_s/i32": [],
+  "f64.convert_u/i32": [],
+  "f64.convert_s/i64": [],
+  "f64.convert_u/i64": [],
+  "f64.promote/f32": [],
+  "i32.reinterpret/f32": [],
+  "i64.reinterpret/f64": [],
+  "f32.reinterpret/i32": [],
+  "f64.reinterpret/i64": []
+};
```
Comment (xtuc) [inline src/compiler/AST/signatures.js]: I think we could generate that like:

```js
valtypes = ['f32', 'i32', ...]
numeric_instructions = valtypes.map(...)
```

File: src/compiler/parsing/watf/number-literals.js [hunk 5]
```diff
@@ -51,6 +52,14 @@ export function parse32I(sourceString: string): number {
   return value;
 }
 
+export function parseU32(sourceString: string): number {
+  const value = parse32I(sourceString);
+  if (value < 0) {
```
Comment (xtuc) [inline src/compiler/parsing/watf/number-literals.js]: I would use `Math.sign(value) !== 1` instead. It's closer to an actual sign check I think and handles cases like NaN.

File: test/compiler/parsing/fixtures/watf/br_if/basic/expected.json [hunk 6]
```diff
@@ -15,12 +15,12 @@
           "id": "br_if",
           "args": [
             {
-              "type": "FloatLiteral",
+              "type": "NumberLiteral",
```
Comment (xtuc) [inline test/compiler/parsing/fixtures/watf/br_if/basic/expected.json]: Yes, this was actually incorrect/regression.

**Comment by xtuc** [comment]
@johnsudaar do you mind fixing the linting warnings (which will be considered as error soon):
```
/home/travis/build/xtuc/js-webassembly-interpreter/test/interpreter/kernel/exec/conversion-instructions.js
  177:50  warning  property `nan`: Property not found in 'object literal'. See line 23   flowtype-errors/show-errors
  177:50  warning  property `nan`: Property not found in 'object literal'. See line 36   flowtype-errors/show-errors
  177:50  warning  property `nan`: Property not found in 'object literal'. See line 49   flowtype-errors/show-errors
  177:50  warning  property `nan`: Property not found in 'object literal'. See line 124  flowtype-errors/show-errors
  177:50  warning  property `nan`: Property not found in 'object literal'. See line 137  flowtype-errors/show-errors
  177:50  warning  property `nan`: Property not found in 'object literal'. See line 150  flowtype-errors/show-errors
  177:50  warning  property `nan`: Property not found in 'object literal'. See line 163  flowtype-errors/show-errors

/home/travis/build/xtuc/js-webassembly-interpreter/src/compiler/AST/index.js
  213:40  warning  FloatLiteral: Could not resolve name                                                                                                                                                                                                                                                     flowtype-errors/show-errors

/home/travis/build/xtuc/js-webassembly-interpreter/src/compiler/parsing/watf/grammar.js
  719:50  warning  Incorrect number of type parameters (expected 1). See line 719                                                                                                                                                                                                                                 flowtype-errors/show-errors
```

**Pull request merged**